### ISSUE TITLE
[BUG] infinite loop 

### ISSUE THREAD

**Issue by monolidth** [agherzan/yubikey-full-disk-encryption#77]
Hey @agherzan,
thanks for this great module. I'm using the `ykfde-suspend.service`. After locking my encrypted volume. I face some weird bugs e.g. 

**Bug 1: The module freeze and no password promt appears**

**Bug 2: Weird infinity loop, which shows "no Entity"**
The prompt appears but this bug appears if the usb device is present (at first) and you remove it in the password prompt.

Either, I have to restart which is kind of annoying. 

I'm using the `5.10.16-arch1-1` kernel.
I also recommend to provide github issue templates, because for me it is hard to tell which information are needed as well. 

Regards,
William

**Comment by Vincent43** [comment]
Hi, thx for the report. Currently ykfde-suspend service has experimental status so such problems like yours are likely to happen.

> Bug 1: The module freeze and no password promt appears

On what kind of hardware it occurs? What manufacturer?

>  The prompt appears but this bug appears if the usb device is present (at first) and you remove it in the password prompt.

Do you mean if you remove yubikey?

**Comment by monolidth** [comment]
Well, I'm running this module on a thinkpad t495s.
Detailed information about the device is covered on the [wiki](https://wiki.archlinux.org/index.php/Lenovo_ThinkPad_T495s) page.

> Hi, thx for the report. Currently ykfde-suspend service has experimental status so such problems like yours are likely to happen.

Yes, I noticed that already, but still I kind ask to help me to fix this issue.

> Do you mean if you remove yubikey?
Yes, I remove the yubikey

Regards,
William

**Comment by Vincent43** [comment]
> Well, I'm running this module on a thinkpad t495s.

There is known issue on thinkpads where it tries to call some firmware files (wifi?) before suspend (which doesn't exist in initramfs) then locks infinitely. IIRC it does this only at first suspend after fresh boot.

Anyway ykfde-suspend module needs to be rewritten.

**Comment by monolidth** [comment]
Meaning that this would make this service pointless? Looks like I should switch to suspend to disk but thanks for giving me this interesting input.

Regards,
William

**Comment by Vincent43** [comment]
No, if the service is fixed then it's not pointless.

**Comment by agherzan** [comment]
Indeed. The suspect work is not robust. If you are willing to invest some time in looking into improving it @monolidth,  we are happy to take a look at your patches. But otherwise, I'd avoid it for now.

**Comment by monolidth** [comment]
@agherzan well, at the moment I don't have any time for my own projects. I will look for different solution.
However, I will keep this issue open.

Regards,
William

**Comment by RubenHoms** [comment]
@azenkwed what certificate are you talking about? Could you expand a bit on how you got this working?
